### PR TITLE
Removed GetUrlWithoutQuery() and GetUrlAuthorityWithScheme() from Azure::Core::Url and added GetScheme().

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/2040

### CODE REVIEW

File: sdk/core/azure-core/src/http/log_policy.cpp [hunk 1]
```diff
@@ -37,13 +37,32 @@ inline void AppendHeaders(
   }
 }
 
+inline std::string GetUrlWithoutQuery(Url url)
```
Comment (antkmsft) [inline sdk/core/azure-core/src/http/log_policy.cpp:40]: This is going to be invoked when logging during each request. Let's use `std::ostringstream` - either take as a parameter (non-const ref), or build in a string stream, and then `return sstream.c_str();`.

File: sdk/core/azure-core/src/http/log_policy.cpp [hunk 2]
```diff
@@ -37,13 +37,32 @@ inline void AppendHeaders(
...
+}
+
 inline std::string GetRequestLogMessage(LogOptions const& options, Request const& request)
 {
   auto const& requestUrl = request.GetUrl();
 
   std::ostringstream log;
   log << "HTTP Request : " << request.GetMethod().ToString() << " "
-      << requestUrl.GetUrlWithoutQuery();
+      << GetUrlWithoutQuery(requestUrl);
```
Comment (antkmsft) [inline sdk/core/azure-core/src/http/log_policy.cpp:65]: Yep - better accept `ostringstream& log` as a parameter for `GetUrlWithoutQuery()`.

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/details/key_serializers.hpp [hunk 3]
```diff
@@ -40,11 +40,26 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys { nam
         KeyVaultKey& key,
         Azure::Core::Json::_internal::json const& json);
 
+    std::string static inline GetUrlAuthorityWithScheme(Azure::Core::Url url)
```
Comment (antkmsft) [inline sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/details/key_serializers.hpp:43]: static before return type.
inline is not needed - functions defined within a class or structs are automatically inline.

**Review by antkmsft — APPROVED** [review]
LGTM, given that you take the feedback

File: sdk/core/azure-core/src/http/log_policy.cpp (hunk 1 above)
Comment (antkmsft) [inline sdk/core/azure-core/src/http/log_policy.cpp:40]: Take value by const ref. We should avoid creating copies.
```suggestion
inline std::string GetUrlWithoutQuery(Url const& url)
```

File: sdk/core/azure-core/src/http/log_policy.cpp (hunk 2 above)
Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/log_policy.cpp:65]: What's the benefit? Right now the log ostringstream change is all in one place, I don't know if it's a good idea to have that be passed to the `GetUrlWithoutQuery()` method.

**Pull request merged**